bindings/pyroot/cppyy/CPyCppyy/CMakeLists.txt

```diff
@@ -54,6 +54,18 @@ set(libname cppyy)
 add_library(${libname} SHARED ${headers} ${sources})
 # Set the suffix to '.so' and the prefix to 'lib'
 set_target_properties(${libname} PROPERTIES  ${ROOT_LIBRARY_PROPERTIES})
+# It is crucial to add the path to the other ROOT libraries for compatibility with pip build backends.
+# After the `pip install` step, the target install user directory will look like:
+# site-packages/
+# -- cppyy/
+# -- cppyy_backend/
+# -- libcppyy.so
+# -- libcppyy_backend.so
+# -- libROOTPythonizations.so
+# -- ROOT/
+# With most of the ROOT libraries being stored in `ROOT/lib`. The RPATH must then be updated for the CPython extension
+# shared libraries.
+set_target_properties(${libname} PROPERTIES  INSTALL_RPATH "${CMAKE_INSTALL_RPATH}:$ORIGIN/ROOT/lib")
```

Comment:
I *think* all of these hardcoded rpaths should now not be necessary if `CMAKE_INSTALL_PYTHONDIR` and `CMAKE_INSTALL_LIBDIR` are set correctly, at least relative to each other.

There are still some quirks with unneeded artifacts when these directories are not identical, which I'm trying to figure out:
https://github.com/root-project/root/pull/19044

But that should not be a problem for the functionality.